### PR TITLE
[ut] add ut for proxy

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add ut for pkg/search/proxy

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
cc @Poor12 for review

File: pkg/search/proxy/store/multi_cluster_cache_test.go [hunk 1]
```diff
@@ -94,7 +111,7 @@ func TestMultiClusterCache_HasResource(t *testing.T) {
 		want bool
 	}{
 		{
-			"has pods",
+			"has gets",
```
Comment (Poor12) [inline pkg/search/proxy/store/multi_cluster_cache_test.go:114]: Why change to has gets？

File: pkg/search/proxy/controller_test.go [hunk 2]
```diff
@@ -242,6 +324,416 @@ func TestController_reconcile(t *testing.T) {
...
+			},
+			want: want{
+				clusterProxying: true,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			karmadaProxying, clusterProxying, cacheProxying = false, false, false
```
Comment (Poor12) [inline pkg/search/proxy/controller_test.go:438]: Why karmadaProxying, clusterProxying, cacheProxying is always false? A little confused.

File: pkg/search/proxy/store/multi_cluster_cache_test.go (hunk 1 above)
Comment (ikaven1024) [reply pkg/search/proxy/store/multi_cluster_cache_test.go:114]: It's a mistake. I will revert it.

File: pkg/search/proxy/controller_test.go (hunk 2 above)
Comment (ikaven1024) [reply pkg/search/proxy/controller_test.go:438]: These variables is to record whether these functions are called. So init to false before each test.

https://github.com/karmada-io/karmada/blob/c8ca014081638fcc8a5eb1e1673c44fbf81bb06a/pkg/search/proxy/controller_test.go#L328-L348

Comment (Poor12) [reply pkg/search/proxy/controller_test.go:438]: Oh, I see.

**Comment by Poor12** [comment]
Generally LGTM
/cc @RainbowMango @jwcesign for details.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**